apps/analysis/translation.py

```diff
@@ -0,0 +1,79 @@
+import json
+
+from apps.teams.utils import current_team
+
+from .const import LANGUAGE_CHOICES
+
+TRANSLATION_PROMPT_TEMPLATE = """Translate these chat messages to {target_lang_name}.
+Return a JSON array where each object has a "translation" field.
+
+### Messages to translate:
+```json
+{message_data}
+```
+
+Output only the JSON array, no additional text."""
+
+
+class TranslationError(Exception):
+    pass
+
+
+def translate_messages_with_llm(messages, target_language, llm_provider, llm_provider_model):
+    """
+    Translate chat messages using the specified LLM provider and model
+    Only translates messages that don't already have the target language translation.
+    """
+    messages_to_translate = []
+    message_indices = []
+
+    for i, msg in enumerate(messages):
+        if not msg.translations:
+            msg.translations = {}
+
+        if target_language not in msg.translations:
+            messages_to_translate.append(msg)
+            message_indices.append(i)
+
+    if not messages_to_translate:
+        return messages
+    try:
+        with current_team(llm_provider.team):
+            llm_service = llm_provider.get_llm_service()
+            model_name = llm_provider_model.name
+            llm = llm_service.get_chat_model(model_name, temperature=0.1)
+            message_data = []
+            for msg in messages_to_translate:
+                message_data.append({"id": str(msg.id), "content": msg.content, "role": msg.role})
+
+            language_names = dict(choice for choice in LANGUAGE_CHOICES if choice[0])
+            target_lang_name = language_names.get(target_language, target_language)
+
+            prompt = TRANSLATION_PROMPT_TEMPLATE.format(
+                target_lang_name=target_lang_name, message_data=json.dumps(message_data)
+            )
+
+            response = llm.invoke(prompt)
+            translated_data = json.loads(response.content)
+
+            for item in translated_data:
+                message_id = item["id"]
+                for message in messages_to_translate:
+                    if str(message.id) == message_id:
+                        message.translations[target_language] = item["translation"]
+                        message.save(update_fields=["translations"])
```

Comment:
It would be more efficient to use `Messages.objects.bulk_update(updated_messages, fields=["translations"])`